fabtests/Makefile.am

```diff
@@ -141,6 +141,7 @@ nobase_dist_config_DATA = \
 	pytest/default/test_shared_ctx.py \
 	pytest/default/test_ubertest.py \
 	pytest/default/test_unexpected_msg.py \
+	pytest/default/test_shm_sighandler.py \
```

Comment:
Since this is shm specific test, should we create `shm` directory and put it there?

Reply:
I had that concern too. But the original test code is already created in `regression` which is not prov specific. Not sure whether we should make this test in `fabtests/prov/shm` instead? @j-xiong

Reply:
I originally thought about putting the test under prov/shm, but decided to go with a new "regression" category to catch all the future regression tests.  I think either way makes sense. @shefty what's your opinion?

Reply:
I would leave it where it is.  The test itself mostly doesn't care what provider we're calling.  It's checking signal intercept handling, which could apply to any provider.  We could probably rename the test and allow passing in the provider name as an argument.  I would guess this test might also apply to EFA, and could also apply to the link and peer providers that use shm.

Reply:
If you think so...efa is also dependent on shm so I thought it'd fit best in the default dir (and it should work on any build system)